### PR TITLE
Listing services classes and plans

### LINKED ISSUE #215: List services classes and plans
Create a Carrier command to list the service classes and plans available in the Kubernetes cluster.

## Acceptance criteria

* Carrier provides a command to list service classes and plans
* The output is humanly readable
* The command is verified by acceptance tests

### PR DESCRIPTION
**Pull request by andreas-kupries**
Fixes #215 

- New command `service-classes`
- New command `service-plans CLASS`
- Extended command `create-service` with command completion of class and plan arguments.

TODO: More acceptance tests.

### CODE REVIEW

**Comment by andreas-kupries** [comment]
Ready for review.

File: internal/services/catalog_service.go [hunk 1]
```diff
@@ -32,6 +32,190 @@ type CatalogService struct {
...
+
+	// spec:
+	//   clusterServiceBrokerName: minibroker
+	//   clusterServiceClassRef:
+	//     name: mongodb
+	//   description: NoSQL document-oriented database that stores JSON-like documents with
+	//     dynamic schemas, simplifying the integration of data in content-driven applications.
+	//   externalID: mongodb-4-0-12
+	//   externalName: 4-0-12
+	//   free: true
```
Comment (jimmykarily) [inline internal/services/catalog_service.go:106]: remove comments?

File: internal/services/catalog_service.go [hunk 2]
```diff
@@ -32,6 +32,190 @@ type CatalogService struct {
...
+	//	  bindingRetrievable: false
+	// **	  clusterServiceBrokerName: minibroker
+	// **	  description: Helm Chart for redis
+	//	  externalID: redis
+	// **	  externalName: redis
+	//	  planUpdatable: false
+	//
+	// Ad *1: This hash is the value of the
+	// 	`servicecatalog.k8s.io/spec.clusterServiceClassRef.name`
+	// label of the assocoiated plans.
```
Comment (jimmykarily) [inline internal/services/catalog_service.go:174]: same (comments)

File: internal/services/catalog_service.go (hunk 1 above)
Comment (andreas-kupries) [reply internal/services/catalog_service.go:106]: I had left them in as example of the structure the code is digging through to get the information.

File: internal/services/catalog_service.go (hunk 2 above)
Comment (andreas-kupries) [reply internal/services/catalog_service.go:174]: I had left them in as example of the structure the code is digging through to get the information.

File: internal/services/catalog_service.go (hunk 1 above)
Comment (andreas-kupries) [reply internal/services/catalog_service.go]: Removed with commit 88d5750c76298888fa30a556e25ae969088be806

File: internal/services/catalog_service.go (hunk 2 above)
Comment (andreas-kupries) [reply internal/services/catalog_service.go]: Removed with commit 88d5750c76298888fa30a556e25ae969088be806

File: cmd/internal/client/delete.go [hunk 3]
```diff
@@ -38,13 +38,17 @@ var CmdDeleteApp = &cobra.Command{
...
-		app, cleanup, _ := paas.NewCarrierClient(cmd.Flags())
+		app, cleanup, err := paas.NewCarrierClient(cmd.Flags())
 		defer func() {
 			if cleanup != nil {
 				cleanup()
 			}
 		}()
 
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveNoFileComp
```
Comment (jimmykarily) [inline cmd/internal/client/delete.go:49]: when we have an error, we return no error?

Comment (andreas-kupries) [reply cmd/internal/client/delete.go:49]: You are in the match code. There is no way to return an error

Comment (andreas-kupries) [reply cmd/internal/client/delete.go:49]: You can only say 'no matches'

**Review by jimmykarily — APPROVED** [review]

**Pull request merged**